src/pages/EmailVerification/EmailVerification.tsx

```diff
@@ -0,0 +1,115 @@
+
+function EmailVerification() {
+
+  const [sent, setSent] = useState(false);
+  const [sending, setSending] = useState(false);
+  const [error, setError] = useState(false);
+  const [verified, setVerified] = useState();
+
+  const handleClick = async () => {
+    try {
+      setSending(true);
+
+      const result = await resendEmail();
+      setSending(false);
+      setSent(result);
+      setError(false);
+
+    } catch {
+      setSending(false);
+      setError(true);
+    }
+  };
+
+  useMount(async () => {
+    const verifyUserReturn = await verifyUser();
+    if (verifyUserReturn) {
+      setVerified(verifyUserReturn);
+    } else {
+      setVerified(false);
+    }
+  });
+
+  const renderNewEmailVerification = () => {
+    if (error) {
+      return (
+        <div className={styles['email-verification']}>
+          <p className={styles['title']}>
+            <RiCloseCircleFill className={styles['error']}/>
+            There was an error sending a verification email to your account.
+          </p>
+          <p className={styles['paragraph']}>
+            Please verify that your account has the correct email address.</p>
+        </div>
+      );
+    }
+    if (sending) {
+      return (
+        <div className={styles['spinner']}>
+          <Spinner
+            animation='border'
+            role='status'
+            variant={'dark'}
+          />
+          <span> Loading...</span>
+        </div>
+      );
+    }
+    if (verified) {
+      // Need to be redirected back to explore page because it has been verified
+      // Reference for script https://stackoverflow.com/questions/3292038/redirect-website-after-specified-amount-of-time
+      return (
+        <div className={styles['email-verification']}>
+          <p className={styles['title']}>
+            You're account has been verified and will be directed shortly.
+          </p>
+          <script>
+            delay = (2000) => {
+            // tslint:disable-next-line:jsx-curly-spacing
+            window.location.href = `/explore`
+          }
+          </script>
+        </div>
+      );
+    }
+    if (sent) {
+      return (
+        <div className={styles['email-verification']}>
+          <p className={styles['title']}>
+            <RiCheckboxCircleFill className={styles['check']}/>
+            A verification link has been sent to your email.
+          </p>
+          <p className={styles['paragraph']}>
+            Please go to your email account and click on the link that has
+            just been sent to you to verify your account</p>
+        </div>
+      );
+    }
+    return (
+      <div className={styles['email-verification']}>
+        <p className={styles['title']}>
+          Your account has not been verified.
+        </p>
+        <p className={styles['paragraph']}>
+          To verify your account please, click the button below to send a
+          verification email to your account. </p>
```

Comment:
The outer wrapper is repeated for every branch of your if statements. Instead do your if statements inside the template


You can use turnery expressions to simplify the templating.
Also, create  a hook to handle the loading and error state management for you.

`util\usePerformRequest.ts`
```
import {useState} from 'react';

export function usePerformRequest() {
  const [{loading, error}, setSending] = useState({error: false, loading: false});
  const performRequest = async (action: () => Promise<any>) => {
    try {
      setSending({loading: true, error: false});
      await action();
      setSending({loading: false, error: false});
    } catch {
      setSending({loading: false, error: true});
    }
  };
  return {performRequest, loading, error};
}
```


```
  function EmailVerification() {
  const [sent, setSent] = useState(false);
  const [verified, setVerified] = useState();
  const {loading, error, performRequest} = usePerformRequest();
  const handleClick = async () => await performRequest(async () => setSent(await resendEmail()));
  useMount(async () => await performRequest(async () => setVerified(await verifyUser() ?? false)));
  return (
    <Layout isEmailVerificationPage={true}>
      {
        loading && !error ?
          <div className={styles['spinner']}>
            <Spinner
              animation='border'
              role='status'
              variant={'dark'}
            />
            <span> Loading...</span>
          </div> :
          <div className={styles['email-verification']}>
            <p className={styles['title']}>
              {error
                ? <RiCloseCircleFill className={styles['error']}/>
                : !verified && sent && <RiCheckboxCircleFill className={styles['check']}/>}
              {error
                ? 'There was an error sending a verification email to your account.' :
                verified
                  ? 'You\'re account has been verified and will be directed shortly.'
                  : sent
                    ? 'A verification link has been sent to your email.'
                    : 'Your account has not been verified.'}
            </p>
            {verified && !error ? <DelayedRedirect path={RoutePaths.explore} delay={2}/> :
              <>
                <p className={styles['paragraph']}>
                  {error
                    ? 'Please verify that your account has the correct email address.'
                    : sent
                      ? `Please go to your email account and click on the link that has
                    just been sent to you to verify your account`
                      : `To verify your account please, click the button below to send a
                    verification email to your account. `}
                </p>
                {!verified && !sent && <Button onClick={handleClick}>Send another verification email</Button>}
              </>
            }
          </div>
      }
    </ Layout>);
}
```

Reply:
I am getting a Cognitive Complexity of functions should not be too high issue.